app/models/bookmarks.rb

```diff
@@ -0,0 +1,19 @@
+class Bookmarks < ActiveRecord::Base
+  validates :user_id, presence: true
+  validates :post_id, presence: true
+
+  validate :entry_does_not_exist, :on => :create
+
+  belongs_to :user
+  belongs_to :post
+
+  def item
+    Bookmarks.where(post_id: post_id, user_id: user_id)
+  end
+
+  def entry_does_not_exist
+    if item.exists?
+      errors[:base] << 'You cannot add the same post twice.'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.